CaloFilters/src/FilterEcalNNTrigger_module.cc

```diff
@@ -0,0 +1,127 @@
+          diagLevel_       (config().diagLevel())
+        {
+           produces<TriggerInfo>();
+        }
+
+        bool filter(art::Event& event) override;
+        void beginJob() override;
+
+
+     private:
+        art::ProductToken<CaloClusterCollection> caloClusterToken_;
+        MVATools          caloBkgMVA_;
+        float             minEtoTest_;
+        float             minMVAScore_;
+        int               diagLevel_;
+
+        bool filterClusters(const art::Handle<CaloClusterCollection>& caloClustersHandle, TriggerInfo& trigInfo);
+  };
+
+
+  void FilterEcalNNTrigger::beginJob()
+  {
+      caloBkgMVA_.initMVA();
+  }
+
+
+
+  bool FilterEcalNNTrigger::filter(art::Event& event)
+  {
+      art::Handle<CaloClusterCollection> caloClustersHandle;
+      event.getByToken(caloClusterToken_, caloClustersHandle);
+      
+      auto trigInfo = std::make_unique<TriggerInfo>();
+      bool retval   = filterClusters(caloClustersHandle, *trigInfo);
+      event.put(std::move(trigInfo));
+      
+      return retval;
+  }
+
+
+  //----------------------------------------------------------------------------------------------------------
+  bool FilterEcalNNTrigger::filterClusters(const art::Handle<CaloClusterCollection>& caloClustersHandle, TriggerInfo& trigInfo)
+  {
+       const Calorimeter& cal = *(GeomHandle<Calorimeter>());
+       const CaloClusterCollection& clusters(*caloClustersHandle);
+ 
+       bool select(false);
+       for (auto clusterIt=clusters.begin(); clusterIt != clusters.end();++clusterIt)
+       {
+          if (clusterIt->energyDep() < minEtoTest_) continue;
+
+          const auto& hits          = clusterIt->caloHitsPtrVector();
+	  const auto& neighborsId   = cal.crystal(hits[0]->crystalID()).neighbors();
+	  const auto& nneighborsId  = cal.crystal(hits[0]->crystalID()).nextNeighbors();
+	            
+	  double e9(hits[0]->energyDep()),e25(hits[0]->energyDep());
+          for (auto hit : hits)
+	  {
+	      if (std::find(neighborsId.begin(),  neighborsId.end(),  hit->crystalID()) != neighborsId.end())  {e9 += hit->energyDep();e25 += hit->energyDep();}
+	      if (std::find(nneighborsId.begin(), nneighborsId.end(), hit->crystalID()) != nneighborsId.end()) {e25 += hit->energyDep();}
+	  }
+
+	  std::vector<float> mvavars(8,0.0);
+	  mvavars[0] = clusterIt->energyDep();
+	  mvavars[1] = clusterIt->cog3Vector().perp();
+	  mvavars[2] = clusterIt->size(); 
+	  mvavars[3] = hits[0]->energyDep();
+	  mvavars[4] = (hits.size()>1) ?  hits[0]->energyDep() + hits[1]->energyDep() : hits[0]->energyDep();
+	  mvavars[5] = e9;
+	  mvavars[6] = e25;
+	  mvavars[7] = clusterIt->diskID();
+	  
+	  float mvaout = caloBkgMVA_.evalMVA(mvavars);
+	  if (mvaout < minMVAScore_) continue;
+
+	  select = true;
+	  //size_t index = std::distance(caloClusters.begin(),clusterIt);
+	  //triginfo->_caloClusters.push_back(art::Ptr<CaloCluster>(caloClustersHandle,index));
+     }
+     
+     return select;
```

Comment:
the logic should be changed to allow making the trigger-matching of all the clusters passing the cut.

Reply:
What do you mean?

Reply:
you can have a look at how it's done here: https://github.com/Mu2e/Offline/blob/master/CaloFilters/src/CaloCosmicCalib_module.cc#L221

Reply:
See your other comment